### PR TITLE
Run intercepted AI calls through normal request preparation and decoding

### PR DESCRIPTION
**Pull request by mmkal**
Intercepted models now exercise normal AI request preparation and response decoding. Handlers receive the prepared request without provider credentials and return a native `Response`, including streaming SSE bodies.

```ts
using interception = await project.ai.intercept(({ request }) =>
  Response.json({ response: "scripted reply" }),
);
```

`intercepted/` is stripped without choosing a provider. Use `intercepted/openai/test-model` when OpenAI preparation matters; lifecycle tests can keep names such as `intercepted/echo`.

`@iterate-com/test-support` groups helpers under its `interceptor` export: `aiTextResponse` for text fixtures, `codemodeBackticksResponse` for agent code, and `installResilientAiInterceptor` for reconnecting test clients. Agent-turn inputs retain typed messages; direct `ai.run` inputs accept arbitrary model bodies.

### Risk map

- Review `workers-ai-transport.ts` and `model-interception.ts` first: real and intercepted requests share preparation and decoding, but provider credentials must never reach the interceptor. Then review `AiRpcTarget.run`, the API tests, and fixture migrations.
- Existing callbacks must read `request.body` and return `Response` objects. Ordinary JSON and streamed SSE delivery/closure are covered; the session-bound installation/reconnect contract is unchanged.
- Public `ai.run` retains JSON decoding, binary/SSE streams, and explicit raw responses. HTTP failures report status and a body excerpt as `Error`, rather than Cloudflare’s private `InferenceUpstreamError` formatting.
- Cap’n Web is unpatched. Interceptors must return fresh, unlocked response bodies; invalid bodies can hang during RPC transfer. Cancellation and exact error-message propagation also retain the limitations below.

### Out of scope / known limitations

Six cases are documented with `createFailing`, not fixed:

| Case | Current behavior | Test file |
| --- | --- | --- |
| Consumer cancels an SSE stream | Producer is not reliably notified | `ai-intercept.itx.e2e.test.ts` |
| Provider disconnects | Producer cancellation is not reliably delivered | `ai-intercept.itx.e2e.test.ts` |
| Producer throws | Reader gets a generic RPC disconnect error instead of the original message | `ai-intercept.itx.e2e.test.ts` |
| Callback returns a consumed response body | Call times out instead of rejecting normally | `ai-intercept.itx.e2e.test.ts` |
| Callback returns a locked response body | Call times out instead of rejecting normally | `ai-intercept.itx.e2e.test.ts` |
| Provider cancellation cleanup never settles | Production AI attempt timeout waits for cleanup | `workers-ai-transport.test.ts` |

- **Producer:** the `ReadableStream` created by the interceptor callback—the code emitting chunks.
- **Provider:** the client/session hosting that interceptor callback, usually the test process. Here it does not mean OpenAI or Cloudflare.
- **Consumer:** the caller reading the response returned by `project.ai.run(...)`.

The last repro was consolidated from #2621. These tests retain the desired assertions and match specific observed failures; RPC waits and reader cleanup in the interception tests are bounded. Use fresh response bodies and normally completed streams while these limitations remain.

Gateway metadata and spend rules remain separate work. Cost events, aggregation, and budget enforcement belong to the future budgeting design rather than this interception contract.

### Validation

- Unpatched local interception API file: 6 passed, 5 expected failures; test execution 21.3 seconds.
- Transport file, including the production-timeout repro: 26 passed, 1 expected failure.
- CI unit tests, lint, typecheck, and formatting pass. Local generated-contract tests also verify the SDK copy matches the OS contract.
- Final unpatched preview passed: 85 browser tests and 213 OS E2E tests, with 11 expected failures and 4 skipped tests in the OS E2E suite. All six app test suites passed.

Codex session: `01a07b03-100a-7803-ba3f-caed48046317`.

| Group | Lines | Significant |
| --- | ---: | ---: |
| Product | +408 -348 | +250 -182 🟩🟩🟥⬜⬜ |
| Tests | +634 -57 | +610 -56 🟩🟩🟩🟩🟩 |
| Config | +36 -2 | +36 -2 🟩⬜⬜⬜⬜ |
| Docs | +118 -16 | +96 -16 🟩⬜⬜⬜⬜ |
| Other | +9 -5 | +7 -4 🟩⬜⬜⬜⬜ |
| Generated | +133 -58 | +52 -6 🟩⬜⬜⬜⬜ |
| **Total** | **+1,338 -486** | **+1,051 -266** 🟩🟩🟩🟩🟥 |

<sub>Lines counts every changed line; Significant ignores blank lines, JS comments, and TypeScript lines with no runtime output. Between a41f579 and bf2bd07, bucketed first-match-wins into the groups defined in `scripts/ci/loc-report.ts`.</sub>

## Environment Config Lease

<details>
<summary>No preview slot recorded.</summary>

| app | status | commit | preview | size (gzip) | deploy duration | test duration | retries | cleanup duration | workflow run | updated | summary |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Auth | released | `bf2bd07` | [https://auth.iterate-preview-8.com](https://auth.iterate-preview-8.com) | 855.5 KiB | 27.5s | 18.8s |  | 5.0s | [Workflow run](https://github.com/iterate/iterate/actions/runs/526719753104976) | 2026-09-11T11:50:26.131Z | Preview app released. |
| Docs | released | `bf2bd07` | [https://docs-preview-8.iterate-dev-preview.workers.dev](https://docs-preview-8.iterate-dev-preview.workers.dev) | 1.10 MiB | 20.0s | 2.2s |  | 4.7s | [Workflow run](https://github.com/iterate/iterate/actions/runs/526719753104976) | 2026-09-11T11:50:25.804Z | Preview app released. |
| Dummy Petshop | released | `bf2bd07` | [https://dummy-petshop.iterate-preview-8.com](https://dummy-petshop.iterate-preview-8.com) | 232.8 KiB | 183ms | 7.1s |  | 1.4s | [Workflow run](https://github.com/iterate/iterate/actions/runs/526719753104976) | 2026-09-11T11:50:25.007Z | Preview app released. |
| OS | released | `bf2bd07` | [https://os.iterate-preview-8.com](https://os.iterate-preview-8.com) | 5.39 MiB (-35.4 KiB vs main) | 61.0s | 213.4s |  | 99.5s | [Workflow run](https://github.com/iterate/iterate/actions/runs/526719753104976) | 2026-09-11T11:52:00.590Z | Preview app released. |
| Semaphore | released | `97b40ec` | [https://semaphore.iterate-preview-8.com](https://semaphore.iterate-preview-8.com) | 453.8 KiB | 22.4s | 23.2s |  | 2.5s | [Workflow run](https://github.com/iterate/iterate/actions/runs/526719753104976) | 2026-09-11T11:50:23.637Z | Preview app released. |
| Streams Example App | released | `bf2bd07` | [https://streams.iterate-preview-8.com](https://streams.iterate-preview-8.com) | 1.57 MiB | 37.7s | 95.1s |  | 5.0s | [Workflow run](https://github.com/iterate/iterate/actions/runs/526719753104976) | 2026-09-11T11:50:26.133Z | Preview app released. |

</details>

---

> [!NOTE]
> **High Risk**
> Touches core LLM egress for agents and `ai.run`, changes a breaking interceptor API, and must keep BYOK credentials out of handler-visible requests while preserving decode semantics across streaming paths.
> 
> **Overview**
> **Intercepted models now go through the same request preparation and response decoding as real provider calls.** Agent turns no longer special-case interception in `agent-llm-request`; `runWorkersAiAttempt` and `itx.ai.run` both build a credential-free prepared `request`, then `sendAiRequest` either dials Workers AI / BYOK gateway or consults the live interceptor.
> 
> **The public interceptor contract changes:** handlers receive `{ source, model, request }` (OpenAI HTTP or Workers AI shapes, no API keys) and must return an unused, unlocked **`Response`** (JSON or SSE). Plain-string / `{ text, usage }` agent-turn returns and verbatim `ai-run` payloads are removed; decoding, chunk streaming, usage normalization, and HTTP error shaping match production.
> 
> **New `@iterate-com/test-support`** packages `installResilientAiInterceptor`, `aiTextResponse`, and `codemodeBackticksResponse` so e2e, Playwright fixtures, and unit harnesses emit provider-shaped SSE without duplicating the old shared export. Docs and generated itx API types reflect the new contract; several e2e/transport tests cover streams, parity between real and intercepted models, and **`createFailing`** pins for RPC stream cancellation / consumed-body edge cases still open.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bf2bd0735b9527630a9039a781ddeaf79dc6f94c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/os/src/domains/agents/workers-ai-transport.ts [hunk 1]
```diff
@@ -104,165 +99,88 @@ export function adaptMessagesForModel(
...
-      .replace(/Requested at: [^"\\]*/g, "Requested at: MASKED")
-      .replace(
-        /- Project: \\"(?:[^"\\]|\\.)*?\\" \(slug [^)]*\)(?: — the project worker\/website serves [^"\\]*)?/g,
-        "- Project: MASKED",
-      )
-      .replace(/"content":"@\d+(?= |\\n)/g, '"content":"@OFFSET')
-      .replace(/\/agents\/[A-Za-z0-9._/-]*/g, "/agents/MASKED")
-      .replace(/([?&](?:signature|sig|expires|exp|token|key)=)[^"&\\\s]+/gi, "$1MASKED")
-      .replace(/"prompt_cache_key":"[^"]*"/g, '"prompt_cache_key":"MASKED"')
-  );
```
[the remark this answers is not in the document]
Comment (mmkal) [reply apps/os/src/domains/agents/workers-ai-transport.ts:263]: 🤖 This is existing production behavior, restored deliberately at the author's request rather than changing cancellation semantics as part of interception. The deadline hang is now pinned separately in #2621 with createFailing, using a non-intercepted provider call. Resolving here as explicitly deferred; no production fix is being reintroduced in this PR.
